src/neo/SmartContract/Native/Tokens/NeoToken.cs

```diff
@@ -131,11 +157,35 @@ protected override void PostPersist(ApplicationEngine engine)
 
             // Distribute GAS for committee
 
-            int index = (int)(engine.Snapshot.PersistingBlock.Index % (uint)ProtocolSettings.Default.CommitteeMembersCount);
+            int m = ProtocolSettings.Default.CommitteeMembersCount;
+            int n = ProtocolSettings.Default.ValidatorsCount;
+            int index = (int)(engine.Snapshot.PersistingBlock.Index % (uint)m);
             var gasPerBlock = GetGasPerBlock(engine.Snapshot);
-            var pubkey = GetCommittee(engine.Snapshot)[index];
+            var committee = GetCommitteeFromCache(engine.Snapshot);
+            var pubkey = committee.ElementAt(index).PublicKey;
             var account = Contract.CreateSignatureRedeemScript(pubkey).ToScriptHash();
             GAS.Mint(engine, account, gasPerBlock * CommitteeRewardRatio / 100);
+
+            // Record the cumulative reward of the voters of committee
+
+            if (ShouldRefreshCommittee(engine.Snapshot.PersistingBlock.Index))
+            {
+                BigInteger voterRewardOfEachCommittee = gasPerBlock * VoterRewardRatio * 100000000L * m / (m + n) / 100; // Zoom in 100000000 times, and the final calculation should be divided 100000000L
+                for (index = 0; index < committee.Count; index++)
```

Comment:
Why do we need this loop?

Reply:
It is mainly used to avoid one user, vote to different committees in one epoch and get multiple `voter` rewards.

Reply:
But we have cached the voting data, haven't we?

Reply:
Yes, voting data will not be changed. Another attack like the following:

```
Bi: the i'th block
E(k): the k'th epoch, which contains { B(21k), B(21k+1), B(21k+2), .....B(21k + 20) } blocks.
Ci(E): the i'th committee in one epoch
Vi(E): the `voterSumRewardPerNEO` of Ci in one epoch

|<----       E(0)            -------->|<----                     E(k = 1)                         --------> |
   B1             B11                     B22, B23              ...           B31 B32 
                    ↑                           ↑                                  |
                    |                           |                                  |
                    |                           |                                  |
            vote C1(E(1))              vote C10(E(1))                         Balance changed
                    |                           |                                  |
                    |                           |                                  |
   
   |<----               V1(E(1))           ---->|

                    |<----                            V10(E(1))              ----> |

If all epoch have the same committee, voter in E(1) will get two `voter` rewards.

`voter` rewards in E(1) = Balance0 * [V1(E1) - V1(E0)] + Balance1 * [V10(E1) - V10(E0))]
```

Reply:
@erikzhang   I think it would be difficult to calculate the reward by blocks without sacrificing more storage, like recording incremental gains for voters of each block's committee.

Reply:
I think we need to do a performance test.

Reply:
Ok, we'll do a performance  test.